### ISSUE TITLE
[Bug]: about:crashparent doesn't display the crash system notification on Android 9 device

### ISSUE THREAD

**Issue by delia-pop** [mozilla-mobile/fenix#26382]
### Steps to reproduce

1. Open a few tabs on Fenix.
2. Tap the URL bar and access about:crashparent.
3. Observe the android notification bar.

### Expected behaviour

As per https://github.com/mozilla-mobile/fenix/issues/26080, when about:crashparent is entered in the URL bar, the browser closes and the crash system notification is displayed.

### Actual behaviour

When about:crashparent is entered in the URL bar, the browser closes, and no crash notification appears in the system notification  bar.  After going about:crashparent a second time, the "Sorry" dialog box is displayed.

### Device name

Xiaomi Redmi Note 8T 

### Android version

Android 9

### Firefox release type

Firefox Nightly

### Firefox version

Latest Nightly 105 from 08/08

### Additional information

https://user-images.githubusercontent.com/89388888/183621355-31d9f370-d435-47c4-adad-9e01a6aace97.mp4

**Comment by mcarare** [comment]
I tested this on a Huawei P10 device with Android 9 (both latest main and store version).

When the app crashes the report crash dialog shows up both when the app is in the foreground or background. 
As I understand it, the notification feature is a backup solution for Android 10+ devices, when the dialog cannot be shown directly.

Is showing the notification on Android 9 devices a requirement?
cc: @cpeterso

**Comment by delia-pop** [comment]
Hi, @cpeterso , @mcarare . Thanks for your input on this. 
I re-tested this with the latest Nightly 109.0a1 from 12/11, Beta 108.0b6 and Firefox RC 108.1.0 with three Android 9 devices:
- With **Google Pixel 3 XL (Android 9)** & **Samsung Galaxy Note 8 (Android 9)** > the crash dialogue is always displayed on home screen right after entering about:crashparent in the URL bar, as expected. 

- With **Xiaomi Redmi Note 8T (Android 9)** > the crash dialogue is **still not displayed after the first attempt**, only after the second (as shown in the initial screen recording).  

Since this seems to be a device specific issue, I'll update the report accordingly, to reflect the problem.

Attaching a new screen recording and also 2 logcats. 

https://user-images.githubusercontent.com/89388888/207049141-47d52d58-8ce7-4fe7-9510-cfafb56cda36.mp4

[Second_Attempt_ForceCrash.txt](https://github.com/mozilla-mobile/fenix/files/10208071/Second_Attempt_ForceCrash.txt)
[First_Attempt_ForceCrash.txt](https://github.com/mozilla-mobile/fenix/files/10208072/First_Attempt_ForceCrash.txt)

**Comment by mcarare** [comment]
Looking at the logs, I can see 

```
2022-12-12 14:42:18.752  1414-5582  com.androi...gerService pid-1414                             D  MIUILOG- Permission Denied Activity : Intent { flg=0x10018000 cmp=org.mozilla.fenix/mozilla.components.lib.crash.prompt.CrashReporterActivity (has extras) } pkg : org.mozilla.fenix uid : 10183 tuid : 10028

2022-12-12 14:42:18.753   642-2010  /vendor/bi....0-service pid-642                              I  destroyService is called for service : org.mozilla.fenix/mozilla.components.lib.crash.handler.CrashHandlerService

```

This means that the app starting the corresponding activity (show crash report prompt) was blocked by the OS. 
For Xiaomi there are permissions labeled as "other permissions" , additional permission tot he Android normal ones. 
Granting the “Display pop-up window” and “Display pop-up while in background” (depending on the MIUI version) should allow our app to show the crash reporter.

**Comment by delia-pop** [comment]
Further tested with our available Xiaomi devices + allowing the "Display pop-up windows while running in the background" permission from device's Settings:

- With Xiaomi Redmi Note 8T (Android 9) & Xiaomi Mi4i (Android 5.0.2) > the crash dialog is now displayed after entering about:crashparent. As Mihai said, granting the said permission "fixed" the issue for Android 5-9 devices. 

- For Xiaomi Pad5 (Android 12), Xiaomi 12 Pro (Android 12) & Xiaomi Mi 8 Lite (Android 10), the crash notifications are not sent, not even after multiple attempts to crash the app.  

Attaching the logcat from Xiaomi 12 Pro (Android 12): 
[A12_Xiaomi.txt](https://github.com/mozilla-mobile/fenix/files/10208999/A12_Xiaomi.txt)


Should we open a separate issue for this?

**Comment by mcarare** [comment]
The Xiaomi 12 Pro (Android 12) logs show:  `MIUILOG- Reject service :Intent`.

AFAIK there is no direct setting in MIUI to enable allowing this, as opposed to the pop-up issue. We would need to write a custom component to ensure the service start is not blocked (the start scheduling to be more precise) or implement a different component for Xiaomi devices ( and maybe others?).

**Comment by cpeterso** [comment]
> The Xiaomi 12 Pro (Android 12) logs show: `MIUILOG- Reject service :Intent`.
> 
> AFAIK there is no direct setting in MIUI to enable allowing this, as opposed to the pop-up issue.

@mcarare, so there is no "Display pop-up windows while running in the background" MIUI permission we can add to Fenix's app manifest? Xiaomi users must manually enable "Display pop-up windows while running in the background" in their MIUI device settings?

I wonder why the second crash popup is shown even if the user didn't enable the "Display pop-up windows while running in the background" setting.

> * For Xiaomi Pad5 (Android 12), Xiaomi 12 Pro (Android 12), Xiaomi Mi11 Lite (Android 11) & Xiaomi Mi 8 Lite (Android 10) > the crash notifications are not sent, not even after multiple attempts to crash the app.
> 
> Should we open a separate issue for this?

@delia-pop That sounds pretty bad. Please open a new issue for that Xiaomi issue on Android 10+.

**Comment by mcarare** [comment]
@cpeterso That setting is only in the MIUI OS of Xiaomi devices, it does not really exist in Android. Not sure why the second one is shown, the logs do not show any difference between the first and the second attempt.

**Comment by delia-pop** [comment]
> @delia-pop That sounds pretty bad. Please open a new issue for that Xiaomi issue on Android 10+.

Submitted https://github.com/mozilla-mobile/fenix/issues/28181.

**Comment by cpeterso** [comment]
Moved to bugzilla: https://bugzilla.mozilla.org/show_bug.cgi?id=1805973

Change performed by the [Move to Bugzilla add-on](https://addons.mozilla.org/en-US/firefox/addon/move-to-bugzilla/).
